portal/models/user.py

```diff
@@ -1671,8 +1670,13 @@ def check_role(self, permission, other_id):
 
         abort(401, "Inadequate role for {} of {}".format(permission, other_id))
 
-    def has_role(self, role_name):
-        return role_name in [r.name for r in self.roles]
+    def has_role(self, role_name_or_list):
```

Comment:
I think if you used [arg unpacking(?)](http://book.pythontips.com/en/latest/args_and_kwargs.html#usage-of-args) you wouldn't need to do any introspection (when called with individual roles) or modify any existing method calls
```suggestion
    def has_role(self, *role_names):
```

Official python docs:
[Arbitrary Argument Lists](https://docs.python.org/dev/tutorial/controlflow.html#arbitrary-argument-lists)

Reply:
excellent suggestion, @ivan-c  - haven't used that trick in years, and seem to have overlooked it.